webpack.config.js

```diff
@@ -1,16 +1,34 @@
-const merge = require('webpack-merge');
+const path = require('path');
 const defaultWebpackConfig = require('./config/webpack/webpack.config');
+const loadSiteConfig = require('./scripts/generate-app-config/loadSiteConfig');
+const TerraDevSite = require('./config/webpack/plugin/TerraDevSite');
 
-const coreConfig = () => ({
-  resolve: {
-    alias: {
-      'terra-dev-site': process.cwd(),
-    },
-  },
-});
+/**
+* Generates the file representing app name configuration.
+*/
+const devSiteConfig = (env = {}, argv = {}) => {
+  const config = defaultWebpackConfig(env, argv);
+  // Load the site configuration.
+  const siteConfig = loadSiteConfig();
 
-const mergedConfig = (env, argv) => (
-  merge(defaultWebpackConfig(env, argv), coreConfig())
-);
+  siteConfig.appConfig = {
+    ...siteConfig.appConfig,
+    subline: 'Extended',
+  };
 
-module.exports = mergedConfig;
+  return {
+    ...config,
+    plugins: config.plugins.slice(0, -1).concat([
```

Comment:
Shouldn't we be able to add configuration to enable the terra-application in terra-dev-site?

Reply:
not by extending the webpack config, only by using the plugin directly, I could have instead extended terra-toolkit's webpack config and added the terra-dev-site plugin to that, but i want to test the provided webpack config here as much as possible.

Reply:
Or i could have added env variables to the existing webpack config but i resisted that because I don't want to encourage continued use of it beyond it's current functionality.